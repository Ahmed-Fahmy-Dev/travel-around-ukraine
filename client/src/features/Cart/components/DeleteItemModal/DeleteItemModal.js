import {
  Button as MuiButton,
  Dialog,
  DialogActions,
  DialogContent,
  DialogContentText,
  DialogTitle,
  styled,
  Typography,
} from '@mui/material';
import React from 'react';
import PropTypes from 'prop-types';

const Button = styled(MuiButton)({
  padding: '12px 24px',
});

const DeleteItemModal = ({ open, onClose, onDelete, tourTitle }) => (
  <Dialog
    open={open}
    onClose={onClose}
    aria-labelledby="delete-dialog-title"
    aria-describedby="delete-dialog-description"
  >
    <DialogTitle id="delete-dialog-title">Delete this tour?</DialogTitle>
    <DialogContent>
      <DialogContentText id="delete-dialog-description">
        Are you sure you want to delete tour:{' '}
        <Typography component="span" fontStyle="italic">
          {tourTitle}
        </Typography>
      </DialogContentText>
    </DialogContent>
    <DialogActions>
      <Button onClick={onClose} disableElevation>
        Cancel
      </Button>
      <Button onClick={onDelete} disableElevation autoFocus>
        Delete
      </Button>
    </DialogActions>
  </Dialog>
);

DeleteItemModal.propTypes = {
  open: PropTypes.bool.isRequired,
  tourTitle: PropTypes.string,
  onClose: PropTypes.func.isRequired,
  onDelete: PropTypes.func.isRequired,
};

DeleteItemModal.defaultProps = {
  tourTitle: '',
};

export default DeleteItemModal;
