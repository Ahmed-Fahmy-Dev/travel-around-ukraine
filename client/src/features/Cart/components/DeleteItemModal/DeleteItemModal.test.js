/* eslint-disable react/prop-types */
import { render } from '@testing-library/react';
import DeleteItemModal from './DeleteItemModal';

jest.mock('@mui/material/Button', () => ({ children }) => <button type="button">{children}</button>);
jest.mock('@mui/material/Dialog', () => ({ children }) => <div>{children}</div>);
jest.mock('@mui/material/DialogActions', () => ({ children }) => <div>{children}</div>);
jest.mock('@mui/material/DialogContent', () => ({ children }) => <div>{children}</div>);
jest.mock('@mui/material/Typography', () => ({ children }) => <p>{children}</p>);
jest.mock('@mui/material/DialogContentText', () => ({ children }) => <div>{children}</div>);
jest.mock('@mui/material/DialogTitle', () => ({ children }) => <h3>{children}</h3>);

describe('DeleteItemModal snapshot test', () => {
  test('should DeleteItemModal match snapshot', () => {
    const onClose = jest.fn();
    const onDelete = jest.fn();

    const { asFragment } = render(<DeleteItemModal open onClose={onClose} onDelete={onDelete} />);

    expect(asFragment()).toMatchSnapshot();
  });
});
