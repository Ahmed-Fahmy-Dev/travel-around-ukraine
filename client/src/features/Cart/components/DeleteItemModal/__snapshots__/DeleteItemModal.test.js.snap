// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DeleteItemModal snapshot test should DeleteItemModal match snapshot 1`] = `
<DocumentFragment>
  <div>
    <h3>
      Delete this tour?
    </h3>
    <div>
      <div>
        Are you sure you want to delete tour: 
        <p />
      </div>
    </div>
    <div>
      <button
        type="button"
      >
        Cancel
      </button>
      <button
        type="button"
      >
        Delete
      </button>
    </div>
  </div>
</DocumentFragment>
`;
